app/controllers/devise_token_auth/sessions_controller.rb

```diff
@@ -144,5 +126,35 @@ def resource_params
       params.permit(devise_parameter_sanitizer.for(:sign_in))
     end
 
+    def current_resource
+      field = (resource_params.keys.map(&:to_sym) & resource_class.authentication_keys).first
+
+      resource = nil
+      if field
+        q_value = resource_params[field]
+
+        if resource_class.case_insensitive_keys.include?(field)
+          q_value.downcase!
+        end
+
+        q = "#{field.to_s} = ? AND provider='email'"
```

Comment:
I'd move this to a method, it'd be easier to override it, like for example: "I want to login with email but my provider is facebook too"